core/src/main/java/io/questdb/cutlass/line/AbstractLineSender.java

```diff
@@ -177,45 +194,180 @@ public CharSink put(char[] chars, int start, int len) {
             if (ptr + len < hi) {
                 Chars.asciiCopyTo(chars, start, len, ptr);
             } else {
-                throw CairoException.instance(0).put("value too long");
+                throw new LineSenderException("value too long. increase buffer size.");
             }
         }
         ptr += len;
         return this;
     }
 
     public AbstractLineSender metric(CharSequence metric) {
-        if (hasMetric) {
-            throw CairoException.instance(0).put("duplicate metric");
+        validateTableName(metric);
+        if (hasTable) {
+            throw new LineSenderException("duplicate table");
+        }
+        if (metric.length() == 0) {
+            throw new LineSenderException("table name cannot be empty");
         }
         quoted = false;
-        hasMetric = true;
+        hasTable = true;
         encodeUtf8(metric);
         return this;
     }
 
+    private void validateColumnName(CharSequence name) {
+        if (!enableValidation) {
+            return;
+        }
+        if (name.length() == 0) {
+            throw new LineSenderException("column name cannot be empty");
+        }
+        for (int i = 0; i < name.length(); i++) {
```

Comment:
is it meant to avoid calling `length()` with each iteration? I would expect JIT to generate exactly the same code. 

I'm quite positive it will be the same with String. but perhaps using abstract `CharSequence` prevents `length()` to be inlined and then the non-idiomatic loop could be faster.